app/sensors/other_sensors/TPH_sensor_BME280.py

```diff
@@ -100,21 +106,12 @@ def read_data(self) -> dict:
             kalman_data_collector = KalmanDataCollector('temperature', 'pressure', 'humidity')
 
             start_time = time.time()
-
             while time.time() - start_time <= self.reading_time:
-                try:
-                    data = self.__get_data()
+                data = self.__get_data()
+                if data:
                     kalman_data_collector.add_data(data)
```

Comment:
Add a type check for the values in `data` to ensure they are numeric. This check can be added either before calling `add_data` for all sensors or within the `kalman_data_collector`.